src/main/java/com/exasol/adapter/document/documentfetcher/files/FilesDocumentFetcher.java

```diff
@@ -58,6 +64,11 @@ public final CloseableIterator<FetchedDocument> run(final ConnectionPropertiesRe
 
     private CloseableIterator<FetchedDocument> readLoadedFile(final FileSegment fileSegment) {
         final RemoteFile remoteFile = fileSegment.getFile();
+        //add the property if csv
+        if(this.fileTypeSpecificDocumentFetcher instanceof CsvDocumentFetcher){
+            CsvDocumentFetcher csvDocumentFetcher = (CsvDocumentFetcher) this.fileTypeSpecificDocumentFetcher;
+        csvDocumentFetcher.setAdditionalConfiguration(this.additionalConfiguration);
```

Comment:
It's a good idea and I also considered it. However since it's CSV specific for now I went with the current approach. If we require it for other document types later this would be the go-to approach for sure.